pkgs/misc/emulators/duckstation/default.nix

```
@@ -0,0 +1,38 @@
+{ lib, mkDerivation, fetchurl, cmake, SDL2, qtbase, wrapQtAppsHook, qttools
+, ninja, git, gtk2, glib }:
+mkDerivation {
+  pname = "duckstation";
+  version = "latest";
+
+  nativeBuildInputs = [ cmake wrapQtAppsHook qttools git ];
+  buildInputs = [ SDL2 qtbase gtk2 ];
+
+  # I didn't find an better solution for gtk dependency.
+  cmakeFlags = [
+    "-DGTK2_GDKCONFIG_INCLUDE_DIR=${gtk2.out}/lib/gtk-2.0/include"
+    "-DGTK2_GLIBCONFIG_INCLUDE_DIR=${glib.out}/lib/glib-2.0/include"
+  ];
+
+  src = fetchurl {
+    # master from 2020-12-29
+    url = "https://github.com/stenzek/duckstation/archive/dfa36e0.tar.gz";
+    sha256 = "11fm3ri8nxwxd417k3c4w43kkcr60icwjzkddf1kfv7z6ljm3wii";
+  };
```

Move this to the top, just after pname/version and before nativeBuildInputs.

```suggestion
  src = fetchFromGitHub {
    owner = "stenzek";
    repo = pname;
    rev = "dfa36e0fa72f08b478f6de82551eb90b7f6361e2";
    sha256 = "..."; # todo - you need to compute this
  };
```